python_modules/libraries/dagster-aws/dagster_aws/ssm/resources.py

```diff
@@ -71,75 +105,78 @@ def example_job():
               # profile as specified in ~/.aws/credentials file
 
     """
-    return construct_ssm_client(
-        max_attempts=context.resource_config["max_attempts"],
-        region_name=context.resource_config.get("region_name"),
-        profile_name=context.resource_config.get("profile_name"),
-    )
+    return SSMResource.from_resource_context(context)
 
 
-tag_shape = Shape(
-    {
-        "key": Field(str, is_required=True, description="Tag key to search for."),
-        "values": Field(
-            [str],
-            is_required=False,
-            description=(
-                "List of tag values to match on. If no values are provided, "
-                "all parameters with the given tag key will be returned. Note: the "
-                "underlying AWS API will throw an exception when more than 10 "
-                "parameters match the tag filter query."
-            ),
-        ),
-    }
-)
+class ParameterStoreTag(Config):
+    key: str = Field(description="Tag key to search for.")
+    values: Optional[List[str]] = Field(deafult=None, description="List")
```

Comment:
spelling. this makes me concerned about how well tested this is

Reply:
`Optional` fields will default to `None` implicitly - I am a bit surprised that `pydantic.Field` doesn't yell about unrecognized kwargs though